### PR TITLE
[#2291] Support the '...' group for alphabetizing fragment spreads

### LINKED ISSUE #2291: Support the fragment spread group when defining alphabetize rule's groups
**Is your feature request related to a problem? Please describe.**

I'd like to be able to use [the alphabetize graphql-eslint rule](https://the-guild.dev/graphql/eslint/rules/alphabetize), and keep all fragment spreads grouped together, preferably at the end of the selection set.

**Describe the solution you'd like**

The rule already supports the `groups` option, which supports any literal string or the `'*'` string for "everything else."
I'd like to be able to configure the rule this way:
```
'@graphql-eslint/alphabetize': [
  'error',
  {
    groups: ['id', '*', '...'],
    selections: ['OperationDefinition', 'FragmentDefinition'],
  },
]
```

So that this selection set:
```
{
  updatedAt
  ...SomeFragmentB
  ...SomeFragmentA
  id
}
```

Would end up being alphabetized this way:
```
{
  id
  updatedAt
  ...SomeFragmentA
  ...SomeFragmentB
}
```

Since [this line](https://github.com/dimaMachina/graphql-eslint/blob/ba9f4aaa29efa09ab9a1c61487a3e525d91d907c/packages/plugin/src/rules/alphabetize.ts#L286) uses the node's name, the `...` part of the spread is gone from the lexical comparison, which results in fragment spreads being sorted as if they were just regular field selections:
```
{
  id
  ...SomeFragmentB
  ...SomeFragmentA
  updatedAt
}
```

**Describe alternatives you've considered**

I've considered supporting regular expressions in the `groups` option, but that seemed like an overkill, even before I realized that `...SomeFragment` is anyways treated as `SomeFragment`.

**Additional context**
Not much to add. I did code up a solution locally, mostly to understand how it ended up sorting items the way it did, but I'm waiting for a discussion on this issue to conclude before moving forward.

### PR DESCRIPTION
**Pull request by yoavsion**
## Description

Adds a group type to the `groups` property of the `alphabetize` rule indicating where fragment spreads should be in a selection set.

Similarly to the `'*'` group, which stands for "everything else," the `'...'` stands for "all fragment spreads."

Fixes #2291

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## How Has This Been Tested?
Added the following unit tests, ran `yarn test` and verified that the error messages match the expected ones:

- [x] should sort selections by group when `...` is at the start
- [x] should sort selections by group when `...` is between
- [x] should sort selections by group when `...` is at the end

## Checklist:

- [x] I have followed the
      [CONTRIBUTING](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md) doc and the
      style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests and linter rules pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments
When working on this change, the display name for a fragment spread was showing `undefined <fragment name>`, which was because there was no name mapping for that `Kind` — so I went ahead and added display names for _all_ `Kind` values, and sorted the map.

### CODE REVIEW

File: packages/plugin/src/utils.ts [hunk 1]
```diff
@@ -126,27 +126,50 @@ export function truthy<T>(value: T): value is Truthy<T> {
   return !!value;
 }
 
-const DisplayNodeNameMap: Record<string, string> = {
-  [Kind.OBJECT_TYPE_DEFINITION]: 'type',
-  [Kind.OBJECT_TYPE_EXTENSION]: 'type',
-  [Kind.INTERFACE_TYPE_DEFINITION]: 'interface',
-  [Kind.INTERFACE_TYPE_EXTENSION]: 'interface',
+const DisplayNodeNameMap: Record<Kind, string> = {
```
Comment (yoavsion) [inline packages/plugin/src/utils.ts:129]: Added display name for all `Kind` enum values, and sorted the list

File: packages/plugin/src/rules/alphabetize/index.ts [hunk 2]
```diff
@@ -412,3 +407,34 @@ export const rule: GraphQLESLintRule<RuleOptions> = {
...
+}
+
+function getName(node: GraphQLESTreeNode<ASTNode>): string {
+  return (
+    ('alias' in node && node.alias?.value) ||
+    //
+    ('name' in node && node.name?.value) ||
+    ''
+  );
+}
```
Comment (dimaMachina) [inline packages/plugin/src/rules/alphabetize/index.ts:440]: refactored a bit for better readability

File: packages/plugin/src/rules/alphabetize/index.ts [hunk 3]
```diff
@@ -203,7 +203,7 @@ export const rule: GraphQLESLintRule<RuleOptions> = {
             selections: selectionsEnum,
             variables: true,
             arguments: [Kind.FIELD, Kind.DIRECTIVE],
-            groups: ['id', '*', 'createdAt', 'updatedAt'],
+            groups: ['id', '*', 'createdAt', 'updatedAt', '...'],
```
Comment (dimaMachina) [inline packages/plugin/src/rules/alphabetize/index.ts:206]: why this preference? I think the start is better?

**Review by dimaMachina — APPROVED** [review]
I refactored a bit duplicated code

I have only 1 comment https://github.com/dimaMachina/graphql-eslint/pull/2293/files#r1841895177

Thanks for contribution

**Pull request merged**